internal/spec/definition.go

```diff
@@ -85,9 +96,54 @@ func ParseDefinitionFile(content []byte) (*Definition, error) {
 		return nil, err
 	}
 
+	for _, scenario := range specDefinition.Scenarios {
+		for _, nrql := range scenario.Tests.NRQLs {
+			err := validateNRQLTestConfig(nrql)
+			if err != nil {
+				return nil, err
+			}
+		}
+	}
```

Comment:
Thanks for moving the definition validation here. Out of curiosity, why did you choose to iterate the scenarios and nrqls instead of leveraging it to the YML unmarshaler by [customizing the unmarshal behavior](https://pkg.go.dev/gopkg.in/yaml.v3?utm_source=godoc#Unmarshaler)?

Reply:
As discussed, the loop is more readable since the yaml unmarshal customization can be a bit tricky.